core/Controller/LostController.php

```diff
@@ -394,9 +521,37 @@ protected function findUserByIdOrMail($input) {
 		});
 
 		if (\count($users) === 1) {
-			return $users[0];
+			return $users[1];
 		}
 
 		throw $userNotFound;
 	}
-}
+
+	/**
+	 * Sets the password reset params.
+	 *
+	 * Users may not change their passwords if:
+	 * - The account is disabled
+	 * - The backend doesn't support password resets
+	 * - The password reset function is disabled
+	 *
+	 * @param IUser $userObj
+	 * @param array $parameters
+	 * @return array
+	 */
+	protected function setPasswordResetParameters(
+		$userObj = null, array $parameters): array {
+
+		if ($parameters['resetPasswordLink'] === 'disabled') {
+			$parameters['canResetPassword'] = false;
+		} else if (!$parameters['resetPasswordLink'] && $userObj !== null) {
```

Comment:
```suggestion
		} else if ($userObj !== null && !$parameters['resetPasswordLink']) {
```

`$userObj !== null ` is the cheaper call.